internal/core/thirdparty/tantivy/rust-array.h

```diff
@@ -62,4 +68,38 @@ struct RustArrayWrapper {
         }
     }
 };
+struct RustResultWrapper {
+    NO_COPY_OR_ASSIGN(RustResultWrapper);
+
+    RustResultWrapper() = default;
+    explicit RustResultWrapper(RustResult result) : result_(result) {
+    }
+
+    RustResultWrapper(RustResultWrapper&& other) noexcept {
```

Comment:
It seems that RustResultWrapper is not a safe type.
The move construction operation on it make a double free.
I prefer to wrap those `ffi` type such as RustResult with unique_ptr to make it safe.